# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1320696055 18000
#      Mon Nov 07 15:00:55 2011 -0500
# Node ID 13287562863b4cab03d226e2555c9e114aa5f1b5
# Parent  a35d381e22b059c2977e753878c6e32dfd637174
load preferences before trying to query them

* ispref.m: Load preferences before query.

diff --git a/scripts/prefs/ispref.m b/scripts/prefs/ispref.m
--- a/scripts/prefs/ispref.m
+++ b/scripts/prefs/ispref.m
@@ -33,16 +33,17 @@
 
 ## Author: jwe
 
 function retval = ispref (group, pref)
 
   if (nargin == 1)
     retval = isfield (loadprefs (), group);
   elseif (nargin == 2)
+    prefs = loadprefs ();
     if (isfield (prefs, group))
       grp = prefs.(group);
       if (ischar (pref) || iscellstr (pref))
         retval = isfield (grp, pref);
       else
         retval = false;
       endif
     else
